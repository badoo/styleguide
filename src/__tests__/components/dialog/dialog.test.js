import React from 'react';
import ReactDOM from 'react-dom';
import { create, act } from 'react-test-renderer';
import 'jest-styled-components';
import Dialog from '../../../components/dialog/dialog';
import Component from '../../../components/component/component';
import componentProps from '../component/component.mockup.json';

describe('Dialog tests:', () => {
    beforeAll(() => {
        ReactDOM.createPortal = jest.fn((element) => {
            return element;
        });
    });

    afterEach(() => {
        ReactDOM.createPortal.mockClear();
    });

    const importedComponentProps = Object.assign({}, componentProps, {
        tests: componentProps.tests.map((test) => {
            test.Component = Component;

            return test;
        }),
    });
    const props = {
        isOpened: false,
        title: 'SpecComponentEmpty',
        content: <Component {...importedComponentProps} />,
        onClose: null,
    };

    const updatedProps = Object.assign({}, props, {
        isOpened: true,
    });

    const updatedPropsWithOnCloseHandler = Object.assign({}, props, {
        onClose: () => null,
    });

    it('Dialog render null', () => {
        let root;

        act(() => {
            root = create(<Dialog {...props} />);
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

    it('Dialog render component', () => {
        let root;

        act(() => {
            root = create(<Dialog {...updatedProps} />);
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

    it('Dialog componentDidUpdate all paths', () => {
        let root;

        act(() => {
            root = create(<Dialog {...props} />);
        });

        act(() => {
            root.update(<Dialog {...updatedProps} />);

            let instance = root.getInstance();

            instance.componentDidUpdate(updatedProps);
        });

        expect(root.toJSON()).toMatchSnapshot();

        act(() => {
            root.update(<Dialog {...props} />);

            let instance = root.getInstance();

            instance.componentDidUpdate(updatedProps);
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

    it('Dialog handleKeydown test', () => {
        let root;

        act(() => {
            root = create(<Dialog {...props} />);
        });

        act(() => {
            root.update(<Dialog {...updatedProps} />);
            const event = { keyCode: 27 };
            let instance = root.getInstance();
            instance.handleKeydown(event);
        });

        act(() => {
            const event = { keyCode: 27 };
            let instance = root.getInstance();

            instance.handleKeydown(event);
        });

        expect(root.toJSON()).toMatchSnapshot();
    });

    it('Dialog onCloseHandler test', () => {
        let root;

        act(() => {
            root = create(<Dialog {...updatedPropsWithOnCloseHandler} />);
        });

        act(() => {
            let instance = root.getInstance();
            instance.onCloseHandler();
        });

        expect(root.toJSON()).toMatchSnapshot();
    });
});
