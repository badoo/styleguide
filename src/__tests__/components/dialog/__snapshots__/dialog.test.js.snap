// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog tests: Dialog componentDidUpdate all paths 1`] = `
.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 16px;
  height: 16px;
  fill: currentColor;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  color: inherit;
}

.c4 > svg {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: inherit;
  width: inherit;
  max-height: inherit;
  height: inherit;
  background: inherit;
  fill: inherit;
}

.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  background-color: #fff;
  width: 100vw;
  height: 100vh;
  overflow-x: hidden;
  overflow-y: auto;
  z-index: 10;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  font-family: sans-serif;
  position: relative;
  padding: 16px;
  background-color: #fafafa;
  color: #282828;
}

.c2 {
  margin-bottom: 0;
  font: 400 16px -apple-system,system-ui,BlinkMacSystemFont,'Segoe UI',Roboto,Ubuntu,'Helvetica Neue','Lucida Grande',sans-serif;
}

.c3 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  right: 16px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  cursor: pointer;
  opacity: 0.5;
  -webkit-transition: opacity 0.2s ease-out;
  transition: opacity 0.2s ease-out;
}

.c3:hover {
  opacity: 1;
}

.c5 {
  padding: 10px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-x: hidden;
  overflow-y: auto;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c14 {
  display: block;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background: #f0f0f0;
  border-radius: 4px 4px 0 0;
  color: #000;
  padding: 0 16px;
}

.styleguide-sandbox .c15:only-child {
  border-radius: 4px;
}

.c17 {
  margin: 0;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  font-size: 16px;
  font-weight: 300;
  font-family: sans-serif;
  padding: 16px 0;
  word-break: break-word;
}

.c18 {
  margin-left: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #202020;
  opacity: 0.5;
  -webkit-transition: opacity 0.1s ease;
  transition: opacity 0.1s ease;
  padding: 0 8px;
}

.c19:hover {
  opacity: 1;
}

.c20 {
  position: relative;
  padding: 20px;
  border: 1px solid #f0f0f0;
  border-top: 0;
  box-shadow: 0 -1px 0 0 #f0f0f0;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c6 {
  padding-top: 32px;
}

.c7 {
  margin: 0 16px;
  font-size: 32px;
  color: #000;
  font-family: sans-serif;
  word-break: break-word;
}

.c8 {
  font-family: sans-serif;
  font-size: 16px;
  color: #a0a0a0;
  margin-bottom: 16px;
  max-width: 50%;
}

.c12 {
  margin-top: 16px;
}

.c9 {
  font-size: 14px;
  font-family: sans-serif;
  padding: 0 16px;
}

.c9 table {
  width: 100%;
  border-collapse: collapse;
  border-radius: 4px;
  text-align: left;
}

.c9 thead tr {
  border-bottom: 1px solid #7000e3;
}

.c9 thead th {
  padding: 0 16px 8px 4px;
}

.c9 tbody tr:nth-child(2n) {
  background: #f0f0f0;
}

.c9 tbody tr td:nth-child(2) {
  font-style: italic;
  color: #303030;
}

.c9 tbody tr td:nth-child(1) {
  font-weight: 600;
}

.c9 tbody tr td:nth-child(4) {
  width: 45%;
}

.c9 td {
  padding: 8px 16px 8px 4px;
}

.c10 + .styleguide-component__props-data {
  margin-top: 16px;
}

.c11 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: sans-serif;
}

.c11::before {
  margin-right: 4px;
  display: block;
  background: #7000e3;
  color: #fff;
  width: 14px;
  height: 14px;
  line-height: 12px;
  text-align: center;
}

.c11::before {
  content: '+';
}

.c13 + .c13 {
  margin-top: 32px;
}

@media screen and (max-width:376px) {
  .c17 {
    -webkit-flex: 1 1 auto;
    -ms-flex: 1 1 auto;
    flex: 1 1 auto;
    overflow: hidden;
    text-overflow: ellipsis;
    word-break: initial;
  }
}

@media screen and (max-width:376px) {
  .c6 {
    padding-top: 16px;
  }
}

@media screen and (min-width:768px) {
  .c7 {
    margin: 0 0 8px 0;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    padding: 0;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      SpecComponentEmpty
    </h1>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div
        className="c4"
        role="presentation"
      >
        <svg
          height="16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 2.41L13.59 1 8 6.59 2.41 1 1 2.41 6.59 8 1 13.59 2.41 15 8 9.41 13.59 15 15 13.59 9.41 8z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="c5"
  >
    <article
      className="c6"
      data-id="component"
    >
      <header
        className="styleguide-component__header"
      >
        <h1
          className="c7"
        >
          Component
        </h1>
        <div
          className="c8"
        >
          description of component
        </div>
      </header>
      <div
        className="c9"
      >
        <div
          className="c10"
        >
          <span
            className="c11"
            onClick={[Function]}
            role="button"
          >
            PROPERTIES
          </span>
        </div>
      </div>
      <div
        className="c12"
      >
        <div
          className="c13 "
        >
          <section
            className="c14"
            data-name="SpecComponentEmpty"
            data-vrt-locator="sandbox"
          >
            <header
              className="c15 c16"
            >
              <h1
                className="c17"
                onClick={[Function]}
                title="SpecComponentEmpty"
              >
                SpecComponentEmpty
              </h1>
              <div
                className="c18"
              >
                <span
                  className="c19"
                  onClick={[Function]}
                >
                  <div
                    className="c4"
                    role="presentation"
                  >
                    <svg
                      height="16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.444 2H2.556C1.696 2 1 2.696 1 3.556v9.333c0 .859.696 1.555 1.556 1.555h3.11V12.89h-3.11V5.11h10.888v7.778h-3.11v1.555h3.11c.856 0 1.556-.7 1.556-1.555V3.556C15 2.696 14.304 2 13.444 2zM8 6.667l-3.111 3.11h2.333v4.667h1.556V9.778h2.333L8 6.667z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  className="c19"
                  onClick={[Function]}
                >
                  <div
                    className="c4"
                    role="presentation"
                  >
                    <svg
                      height="16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1 12h3v3h2v-5H1v2zm3-8H1v2h5V1H4v3zm6 11h2v-3h3v-2h-5v5zm2-11V1h-2v5h5V4h-3z"
                      />
                    </svg>
                  </div>
                </span>
              </div>
            </header>
            <div
              data-vrt-locator="sandbox-content"
            >
              <div
                className="c20"
              >
                <article
                  className="c6"
                >
                  <header
                    className="styleguide-component__header"
                  >
                    <h1
                      className="c7"
                    />
                    <div
                      className="c8"
                    />
                  </header>
                </article>
              </div>
            </div>
          </section>
        </div>
        <div
          className="c13 "
        >
          <section
            className="c14"
            data-name="SpecComponentPropTypes"
            data-vrt-locator="sandbox"
          >
            <header
              className="c15 c16"
            >
              <h1
                className="c17"
                onClick={[Function]}
                title="SpecComponentPropTypes"
              >
                SpecComponentPropTypes
              </h1>
              <div
                className="c18"
              >
                <span
                  className="c19"
                  onClick={[Function]}
                >
                  <div
                    className="c4"
                    role="presentation"
                  >
                    <svg
                      height="16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.444 2H2.556C1.696 2 1 2.696 1 3.556v9.333c0 .859.696 1.555 1.556 1.555h3.11V12.89h-3.11V5.11h10.888v7.778h-3.11v1.555h3.11c.856 0 1.556-.7 1.556-1.555V3.556C15 2.696 14.304 2 13.444 2zM8 6.667l-3.111 3.11h2.333v4.667h1.556V9.778h2.333L8 6.667z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  className="c19"
                  onClick={[Function]}
                >
                  <div
                    className="c4"
                    role="presentation"
                  >
                    <svg
                      height="16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1 12h3v3h2v-5H1v2zm3-8H1v2h5V1H4v3zm6 11h2v-3h3v-2h-5v5zm2-11V1h-2v5h5V4h-3z"
                      />
                    </svg>
                  </div>
                </span>
              </div>
            </header>
            <div
              data-vrt-locator="sandbox-content"
            >
              <div
                className="c20"
              >
                <article
                  className="c6"
                >
                  <header
                    className="styleguide-component__header"
                  >
                    <h1
                      className="c7"
                    />
                    <div
                      className="c8"
                    />
                  </header>
                </article>
              </div>
            </div>
          </section>
        </div>
        <div
          className="c13 "
        >
          <section
            className="c14"
            data-name="SpecComponentFull"
            data-vrt-locator="sandbox"
          >
            <header
              className="c15 c16"
            >
              <h1
                className="c17"
                onClick={[Function]}
                title="SpecComponentFull"
              >
                SpecComponentFull
              </h1>
              <div
                className="c18"
              >
                <span
                  className="c19"
                  onClick={[Function]}
                >
                  <div
                    className="c4"
                    role="presentation"
                  >
                    <svg
                      height="16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.444 2H2.556C1.696 2 1 2.696 1 3.556v9.333c0 .859.696 1.555 1.556 1.555h3.11V12.89h-3.11V5.11h10.888v7.778h-3.11v1.555h3.11c.856 0 1.556-.7 1.556-1.555V3.556C15 2.696 14.304 2 13.444 2zM8 6.667l-3.111 3.11h2.333v4.667h1.556V9.778h2.333L8 6.667z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  className="c19"
                  onClick={[Function]}
                >
                  <div
                    className="c4"
                    role="presentation"
                  >
                    <svg
                      height="16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1 12h3v3h2v-5H1v2zm3-8H1v2h5V1H4v3zm6 11h2v-3h3v-2h-5v5zm2-11V1h-2v5h5V4h-3z"
                      />
                    </svg>
                  </div>
                </span>
              </div>
            </header>
            <div
              data-vrt-locator="sandbox-content"
            >
              <div
                className="c20"
              >
                <article
                  className="c6"
                >
                  <header
                    className="styleguide-component__header"
                  >
                    <h1
                      className="c7"
                    />
                    <div
                      className="c8"
                    />
                  </header>
                </article>
              </div>
            </div>
          </section>
        </div>
      </div>
    </article>
  </div>
</div>
`;

exports[`Dialog tests: Dialog componentDidUpdate all paths 2`] = `null`;

exports[`Dialog tests: Dialog handleKeydown test 1`] = `null`;

exports[`Dialog tests: Dialog onCloseHandler test 1`] = `null`;

exports[`Dialog tests: Dialog render component 1`] = `null`;

exports[`Dialog tests: Dialog render null 1`] = `null`;
